### PR TITLE
move retry logic to shell script

### PR DESCRIPTION
**Pull request by niranjan92**
https://github.com/Shippable/admiral/issues/1110

### CODE REVIEW

File: common/scripts/docker/installDb.sh [hunk 1]
```diff
@@ -5,28 +5,7 @@ export TIMEOUT=180
...
-      __process_msg "Waiting for database to start"
-      let "counter = $counter + $interval"
-      sleep $interval
-    fi
-  done
-  if [ $db_booted = false ]; then
-    __process_msg "Failed to boot database container"
-    exit 1
-  fi
+  __check_service_up $DB_IP $DB_PORT "database" "180"
```
Comment (ric03uec) [inline common/scripts/docker/installDb.sh]: why are we hardcoding `database`? the env `COMPONENT` should already be there. we need to first look at the data that's already available before adding hardcoded stuff like this which degrades the system

File: common/scripts/lib/_helpers.sh [hunk 2]
```diff
@@ -506,6 +506,56 @@ __copy_script_remote() {
   copy_cmd_out=$(eval $copy_cmd)
 }
 
+# accepts arguments $host $port $serviceName $timeout
+# $timeout is optional and defaults to 60 seconds if not defined
+__check_service_up() {
```
Comment (ric03uec) [inline common/scripts/lib/_helpers.sh]: this should be `__check_service_connection`

File: common/scripts/lib/_helpers.sh [hunk 3]
```diff
@@ -506,6 +506,56 @@ __copy_script_remote() {
...
+    exit 1
+  fi
+  local service=$3
+  if [ -z "$service" ]; then
+    echo "got empty service name in __check_service_up(), exiting..."
+    exit 1
+  fi
+  local interval=3
+  local counter=0
+  local db_booted=false
```
Comment (ric03uec) [inline common/scripts/lib/_helpers.sh]: wrong variable name. make it generic.

File: common/scripts/docker/installDb.sh (hunk 1 above)
Comment (niranjan92) [reply common/scripts/docker/installDb.sh]: had hard coded name because it was being used in database specific script `installDB.sh`. I have updated it to use `$COMPONENT`

File: common/scripts/docker/installDb.sh [hunk 4]
```diff
@@ -4,29 +4,8 @@ export DB_IMAGE="drydock/postgres:$RELEASE"
...
-      let "counter = $counter + $interval"
-      sleep $interval
-    fi
-  done
-  if [ $db_booted = false ]; then
-    __process_msg "Failed to boot database container"
-    exit 1
-  fi
+  __process_msg "Checking $COMPONENT container status on: $DB_IP:$DB_PORT"
+  __check_service_up $DB_IP $DB_PORT $COMPONENT "180"
```
Comment (ric03uec) [inline common/scripts/docker/installDb.sh]: - double quote all the variables being passed around
- set the 180 value at the top in a variable that shoudl be passed in the function call

Comment (ric03uec) [reply common/scripts/docker/installDb.sh]: function name is still incorrect.?? how is this even being tested ? im pretty sure this PR wasnt tested in any way whatsoever

File: common/scripts/lib/_helpers.sh [hunk 5]
```diff
@@ -506,6 +506,56 @@ __copy_script_remote() {
   copy_cmd_out=$(eval $copy_cmd)
 }
 
+# accepts arguments $host $port $serviceName $timeout
+# $timeout is optional and defaults to 60 seconds if not defined
+__check_service_connection() {
+
+  if [ "$#" != "3" ] && [ "$#" != "4" ]; then
+    echo "invalid arguments to __check_service_connection()"
```
Comment (ric03uec) [inline common/scripts/lib/_helpers.sh]: why are we not using __process_error and __process_msg here like the rest of the file?

Comment (niranjan92) [reply common/scripts/lib/_helpers.sh]: @ric03uec 
tried using that but it was giving error - 
```
47ff0ededb29b920037c3070ef8eb85d305193e3d73109261d2add2023e31c24
|___ Checking db container status on: localhost:5432
/home/niranjan/shippable/admiral/common/scripts/lib/_helpers.sh: line 548: process_msg: command not found
```

should i source `logger.sh` in `helper.sh`?

File: common/scripts/lib/_helpers.sh [hunk 6]
```diff
@@ -6,6 +6,7 @@
 # TODO: break up this file into smaller, logically-grouped
 # files after we add release and re-install features
 
+source "$LIB_DIR/_logger.sh"
```
Comment (ric03uec) [inline common/scripts/lib/_helpers.sh]: why is this needed?

Comment (niranjan92) [reply common/scripts/lib/_helpers.sh]: removed the source 
able to run installer correctly 
```
# 14:08:44 #######################################
# Booting database
##################################################
|___ Running database container
|___ Running a new database container
29264c9558140e0cd3e9f39bfb22d41a1f29673d4aa74d4a14bac6ecd6290aa7
|___ Checking db container status on: localhost:5432
|___ Waiting for db to start
|___ db found
|___ Database container successfully running
```

**Comment by ric03uec** [comment]
- can you mention the test steps here. i need to understand what scenarios were tried out.
- this logic will be used for all service retries, so we need to make sure all cases are covered here

**Comment by niranjan92** [comment]
@ric03uec  tested the retry function locally for following -
- empty arguments for $host, $port, $serviceName and correct number of arguments. 
- $timeout should default to 60 seconds if not defined
- checked retry logic by stopping database container locally and calling the function. retry function works as expected. 
- changed logic for remaining services and able to initialize admiral.
